### ISSUE TITLE
New dataset `region_isos_demo` 

### ISSUE THREAD

**Issue by jdhoffa** [2DegreesInvesting/r2dii.data#60]
* This dataset should be an exact duplicate of `region_isos` but with the column `source` equal to "demo_2020". 

* and filter `region` for c("global", "europe",...) plus a handful of sensible regions).

**Comment by maurolepore** [comment]
Add `region_isos_demo`

* Starts not from a .csv but from the exported dataset `region_isos`.
* [ ] Add "data-raw/data_dictionary/region_isos_demo.csv" 
* [ ] In "data-raw/region_isos_demo.R", add something like: 
  ```
    region_isos_demo <- r2dii.data::region_isos %>%
       # code
  use_data(region_isos_demo, overwrite = TRUE)
  ```
* [ ] `usethis::use_r("region_isos_demo")`. Document `region_isos_demo` 
* [ ] `usethis::use_test("region_isos_demo")`. Add regression tests of `region_isos_demo` 
* [ ] `source("data-raw/data_dictionary.R"); devtools::load_all()` 
* [ ] `usethis::use_test("data_dictionary")`. Test `data_dictionary` includes `region_isos_demo`

**Comment by maurolepore** [comment]
Closed by #62
